authority/provisioner/collection.go

```diff
@@ -127,6 +127,8 @@ func (c *Collection) LoadByCertificate(cert *x509.Certificate) (Interface, bool)
 				return c.Load("aws/" + string(provisioner.Name))
 			case TypeGCP:
 				return c.Load("gcp/" + string(provisioner.Name))
+			case TypeACME:
+				return c.Load(string(provisioner.Name))
```

Comment:
Here we are loading by id, I don't think we are currently enforcing unique names, only unique ids, so I think we should follow the logic used in aws or gcp and use `acme/name`.